assets/app/scripts/controllers/overview.js

```diff
@@ -40,22 +53,156 @@ angular.module('openshiftConsole')
+          if (ls.matches(pod)) {
+            services.push(svcName);
+            $scope.podsByService[svcName][name] = pod;
+
+            var isInDepInSvc = false;
+            // Not using angular.forEach so that we can break out early
+            for (var i = 0; !isInDepInSvc && i < deployments.length; i++) {
+              var depName = deployments[i];
+              if ($scope.deploymentsByService[svcName][depName]) {
+                isInDepInSvc = true;
+              }
+            }
+
+            if (!isInDepInSvc) {
+              $scope.monopodsByService[svcName] = $scope.monopodsByService[svcName] || {};
+              $scope.monopodsByService[svcName][name] = pod;
+            }
+          }
+        });
+        if (deployments.length == 0 && services.length == 0 && showMonopod(pod)) {
+          $scope.monopodsByService[""][name] = pod;
+        }
+      });
+
+      console.log("podsByDeployment", $scope.podsByDeployment);      
+      console.log("podsByService", $scope.podsByService); 
+      console.log("monopodsByService", $scope.monopodsByService); 
+    };
+
+    // Filter out monopods we know we don't want to see
+    var showMonopod = function(pod) {
+      // Hide pods in the Succeeded or Terminated phase since these are run once pods
+      // that are done
+      if (pod.status.phase == 'Succeeded' || pod.status.phase == 'Terminated') {
+        return false;
+      } 
+      // Hide our build pods since we are already showing details for currently running or recently
+      // run builds under the appropriate areas
+      for (var id in $scope.builds) {
+        if ($scope.builds[id].podName == pod.metadata.name) {
+          return false;;
+        }
+      }
+      // Hide our deployer pods since it is obvious the deployment is happening when the new deployment
+      // appears.
+      if (pod.metadata.annotations && pod.metadata.annotations.deployment) {
+        return false;
+      }
+
+      return true;
+    };
+
+    var deploymentConfigsByService = function() {
+      $scope.deploymentConfigsByService = {"": {}};
+      angular.forEach($scope.deploymentConfigs, function(deploymentConfig, depName){
+        var foundMatch = false;
+        var depConfigSelector = new LabelSelector(deploymentConfig.template.controllerTemplate.replicaSelector);
+        angular.forEach($scope.services, function(service, name){
+          $scope.deploymentConfigsByService[name] = $scope.deploymentConfigsByService[name] || {};
+
+          var serviceSelector = new LabelSelector(service.spec.selector);
+          if (serviceSelector.covers(depConfigSelector)) {
+            $scope.deploymentConfigsByService[name][depName] = deploymentConfig;
+            foundMatch = true;
+          }
+        });
+        if (!foundMatch) {
+          $scope.deploymentConfigsByService[""][depName] = deploymentConfig;
+        }
       });
+    };
+
+    var deploymentsByService = function() {
+      $scope.deploymentsByService = {"": {}};
```

Comment:
you wouldn't entertain referring to these locally as `bySvc` and `bySvcByDepCfg`, would you?